siteorigin-panels.php

```diff
@@ -663,6 +663,12 @@ static function get_widget_instance( $class_or_hash ) {
 	 */
 	public function activation_redirect( $plugin ){
 		if( $plugin == plugin_basename( __FILE__ ) ) {
+
+			// Prevent redirect in certain situations
+			if ( wp_doing_ajax() || is_network_admin() || isset( $_GET['activate-multi'] ) ) {
```

Comment:
This doesn't work for me. In which circumstances is `activate-multi` used? I don't see it when activating multiple plugins from the plugins page.

As far as I can tell, WP sets `$_POST['action']` to `'activate'` when a single plugin is activated and sets it to `'activate-selected'` when multiple plugins are activated.

Reply:
>  I don't see it when activating multiple plugins from the plugins page.

That's when `activate-multi` is used.  I'm not too sure why it's not triggering for you but it's definitely meant to. [Here's where WordPress uses this](https://github.com/WordPress/WordPress/blob/master/wp-admin/plugins.php#L126). Does disabling all non-SO plugins help? (there might be something interfering)

Reply:
I've identified why this is happening. [This redirect](https://github.com/WordPress/WordPress/blob/master/wp-admin/plugins.php#L126) is done _after_ the plugins are activated. This results in `activate-multi` not existing until after  the `activated_plugin` hook is used.

I can't see any other method of preventing this without using two hooks (`activated_plugin` `and admin_init` - this can be seen in my first commit). Do you see any other ways?

Reply:
Does my previous suggestion not work for you?

Reply:
Sorry for the delay - lost track of this reply.

It doesn't, no. I applied the suggested change to a dev version and a release version (to rule out an issue with the dev version), and it didn't work for either.

Did it work for you?

Reply:
Sorry about the runaround, I must have missed something in my initial checking. I checked the fix using a transient again and that's working for me now. :confounded:  SO I think reverting to that is the best way to go for this.

Reply:
No worries. I think there was something wrong with my previous dev install as the current commit seriously did work so I guess everything evens out.

Just reverted the last commit.